templates/cli/lib/commands/run.js.twig

```diff
@@ -0,0 +1,599 @@
+                await dockerStart(func, variables, port);
+            } else {
+                log('Hot-swapping function files ...');
+                
+                const functionPath = path.join(process.cwd(), func.path);
+                const hotSwapPath = path.join(functionPath, '.appwrite/hot-swap');
+                const buildPath = path.join(functionPath, '.appwrite/build.tar.gz');
+
+                // Prepare temp folder
+                if (!fs.existsSync(hotSwapPath)) {
+                    fs.mkdirSync(hotSwapPath, { recursive: true });
+                } else {
+                    fs.rmSync(hotSwapPath, { recursive: true, force: true });
+                    fs.mkdirSync(hotSwapPath, { recursive: true });
+                }
+
+                await tar
+                    .extract({
+                        gzip: true,
+                        sync: true,
+                        cwd: hotSwapPath,
+                        file: buildPath
+                    });
+
+                const ignorer = ignore();
+                ignorer.add('.appwrite');
+                if (func.ignore) {
+                    ignorer.add(func.ignore);
+                }
+
+                const filesToCopy = getAllFiles(functionPath).map((file) => path.relative(functionPath, file)).filter((file) => !ignorer.ignores(file));
+                for(const f of filesToCopy) {
+                    const filePath = path.join(hotSwapPath, f);
+                    if (fs.existsSync(filePath)) {
+                        fs.rmSync(filePath, { force: true });
+                    } 
+
+                    const fileDir = path.dirname(filePath);
+                    if (!fs.existsSync(fileDir)) {
+                        fs.mkdirSync(fileDir, { recursive: true });
+                    }
+
+                    const sourcePath = path.join(functionPath, f);
+                    fs.copyFileSync(sourcePath, filePath);
+                }
+
+                await tar
+                    .create({
+                        gzip: true,
+                        sync: true,
+                        cwd: hotSwapPath,
+                        file: buildPath
+                    }, ['.']);
+                
+                fs.rmSync(hotSwapPath, { recursive: true, force: true });
+
+                await dockerStart(func, variables, port);
+            }
+        } catch(err) {
+            console.error(err);
+        } finally {
+            Queue.unlock();
+        }
+    });
+}
+
+const run = new Command("run")
+    .alias("dev")
+    .description(commandDescriptions['run'])
+    .configureHelp({
+        helpWidth: process.stdout.columns || 80
+    })
+    .action(actionRunner(async (_options, command) => {
+        command.help();
+    }));
+
+run
+    .command("function")
+    .alias("functions")
+    .description("Run functions in the current directory.")
+    .option(`--functionId <functionId>`, `Function ID`)
+    .option(`--port <port>`, `Local port`)
```

Comment:
lets fix a default port so user doesnt need to enter it

Reply:
It's optional param. If not provided ,question is asked (with default 3000 as placeholder)